presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -1856,6 +1856,10 @@ public void dropView(ConnectorSession session, SchemaTableName viewName)
         return metastore.getTable(new HiveIdentity(session), viewName.getSchemaName(), viewName.getTableName())
                 .filter(ViewReaderUtil::canDecodeView)
                 .map(view -> {
+                    if (!translateHiveViews && !isPrestoView(view)) {
```

Comment:
Can we please have some test for that?

Reply:
@kokosing since this is not a session property, I think it's currently hard to test in the product tests without adding a separate config file. But I'm not sure a different configuration is worth the overhead in product test framework. 

Do you have a suggestion on it? or should we add a session property instead?

Reply:
Do we need product tests for that? If we could test it "unit" tests then it would be nice. I guess it might be possible to enhance file based metastore for that. WDYT?

Reply:
@kokosing added a test in `AbstractTestHive`, the test seems a bit hacky, since we need to create a table through the client (non-presto code path). But I think it should work for our purpose.